### ISSUE TITLE
Failing to provision EC2 with non-trivial resources

### ISSUE THREAD

**Issue by lguminski** [hashicorp/packer#925]
I had to automate the process of creation of AMIs with 120GB of resources, and realized that the current implementation of file provisioner is quite unstable.

This is basically what I want to achieve:

```
    {
      "type": "file",
      "source": "resources",
      "destination": "/mnt"
    },
```

Here is a short list of problems that I faced:
1. First it turned out that the implementation copies each file to /tmp. I have there just several GBs of space, so the system choked on a file which couldn't fit in.
2. I increased space under /tmp, but then I was bitten by #785 (panic: Failed to write data, buffer full for stream 6) - my gist [here](https://gist.github.com/lguminski/9328259#file-gistfile1-txt)
3. So I archived entire directory (120GB) and split it into equally sized chunks (1GB). It was progressing fine, but then Packer always was stuck around 60GB. It could hang like this for several hours. I have checked, copying was not progressing.

As a workaround, in parallel to packer I am starting a process which copies data with rsync, right after instance is started by packer. While copying, packer checks periodically (with a script) if all data is there.  Once done, packer resumes: it stops the instance, and continues with image creation.

So it is working for me now, but that would be great if packer could handle such resources, without any workarounds.
